src/Servers.jl

```diff
@@ -410,104 +307,79 @@ function handle_connection(f::Function, c::Connection;
     return
 end
 
+"creates an async thread that waits a specified amount of time before closing the connection"
+function check_readtimeout(c, readtimeout, wait_for_timeout)
+    @async while wait_for_timeout[]
+        if inactiveseconds(c) > readtimeout
+            @warn "Connection Timeout: $c"
+            try
+                writeheaders(c.io, Response(408, ["Connection" => "close"]))
+            finally
+                close(c)
+            end
+            break
+        end
+        sleep(8 + rand() * 4)
+    end
+    return
+end
 
 """
-Create a `HTTP.Stream` and parse the Request headers from a `HTTP.Transaction`.
-If there is a parse error, send an error Response.
-Otherwise, execute stream processing function `f`.
+Transaction handler: creates a new Stream for the Transaction, calls startread on it,
+then dispatches the stream to the user-provided handler function. Catches errors on all
+IO operations and closes gracefully if encountered.
 """
-function handle_transaction(f::Function, t::Transaction;
-                            final_transaction::Bool=false,
```

Comment:
Per previous comment, I'd prefer to keep the verbose `final_transaction` name as a kw arg.

Reply:
Changed back.